tests/FSharp.Compiler.ComponentTests/Conformance/DelegateTypes/DelegateDefinition.fs

```diff
@@ -30,8 +31,10 @@ namespace FSharpTest
         """
         |> compile
         |> shouldFail
-        |> withErrorCode 552
-        |> withErrorMessage "Only class types may take value arguments"
+        |> shouldFail
+        |> withDiagnostics [
+            (Error 552, Line 3, Col 11, Line 3, Col 13, "Only class types may take value arguments")
```

Comment:
@0101 Error 552 It is used in other places I believe. So adding an extra text will affect the other places.  We could create another error number with a dedicated error text ?

Reply:
See the comments in the linked PR https://github.com/dotnet/fsharp/pull/13078 to see what was agreed there regarding the error message :)

Reply:
Yeah it would be a new error message for this particular case. 

But it's not really related to making the ranges accurate - it just occurred to me when I saw it. But could be done as a separate improvement.

Reply:
Agreed 😀